src/engine/graphics/texture_atlas.cpp

```diff
@@ -45,14 +45,18 @@ void TextureAtlas::AddImage(const std::string &path, bool absolute) {
   } else {
     snprintf(full_path, sizeof(full_path) - 1, "%s", u_find2(path.c_str(), supported_image_formats, false));
   }
+
   auto* img = static_cast<PLImage *>(u_alloc(1, sizeof(PLImage), true));
   if(!plLoadImage(full_path, img)) {
-    Error("Failed to load image (%s)!\n", plGetError());
+    u_free(img);
+    LogWarn("Failed to load image (%s)!\n", plGetError());
```

Comment:
Don't throw the error message here, let the caller deal with that when it returns false. This way we don't get a warning message every time a map gets loaded.